File: demopos-api/DemoPos.Api/Constants/DiscountTypes.cs
namespace DemoPos.Api.Constants;

public static class DiscountTypes
{
    public const string Fixed      = "fixed";
    public const string Percentage = "percentage";
    public const string Percent    = "percent";    // Purchase domain
    public const string Override   = "override";
    public const string Member     = "member";     // Zero-total (prepaid trial card) redemption
}

File: demopos-frontend/src/constants/discountTypes.js
/** Discount type values — used in both UI state and API payloads. */
export const DISCOUNT_TYPE = {
  FIXED:      'fixed',
  PERCENT:    'percent',      // UI-internal value (checkout modal state)
  PERCENTAGE: 'percentage',   // API/backend value (ProductService, VoucherService)
  OVERRIDE:   'override',     // Voucher override type
  MEMBER:     'member',       // Zero-total member redemption
}
